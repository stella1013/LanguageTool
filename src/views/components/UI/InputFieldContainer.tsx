import * as React from 'react';

interface LocalState {
	showInput: boolean;
}
class InputFieldContainer extends React.Component<any, LocalState> {
	constructor(props: any) {
		super(props);

		this.state = {
			showInput: false
		};
	}

	inputVisibleToggle = (e: React.MouseEvent) => {
		if (this.state.showInput === false) {
			this.setState({ showInput: true });
		}
	}

	render() {
		return (
			<div className="inputWrapper">
				{this.state.showInput ? (
					<input
						className="input-element"
						{...this.props.elementConfig}
						value={this.props.value}
						onChange={this.props.changed}
						id={this.props.id}
					/>
				) : (
					<div
						className="current-location"
						onClick={(e: React.MouseEvent) => this.inputVisibleToggle(e)}
					>
						Current Location
					</div>
				)}
			</div>
		);
	}
}

export default InputFieldContainer;
