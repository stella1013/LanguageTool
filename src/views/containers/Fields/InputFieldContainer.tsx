import * as React from 'react';


interface LocalState {
	showInput: boolean;
}
class InputFieldContainer extends React.Component<any, LocalState> {
	constructor(props: any) {
		super(props);

		this.state = {
			showInput: true
		};
	}
	/*
	static getDerivedStateFromProps: React.GetDerivedStateFromProps<any, LocalState> = (props, state) => {
		if (props.showInput !== state.showInput) {
		  return {
			showInput:props.showInput
		  };
		}
		return null;
	  }
	  */
	inputVisibleToggle = (e: React.MouseEvent) => {
		if (this.state.showInput === false) {
			this.setState({ showInput: true });
		}
	}

	render() {
		return (
			<div className="inputWrapper">
				{this.state.showInput ? (
					<input
						className={"input-"+this.props.id}
						{...this.props.elementConfig}
						value={this.props.value}
						onChange={this.props.changed}
						id={this.props.id}
						placeholder={this.props.placeholder}
					/>
				) : (
					<div
						className="current-location"
						onClick={(e: React.MouseEvent) => this.inputVisibleToggle(e)}
					>
						Current Location
					</div>
				)}
			</div>
		);
	}
}

export default InputFieldContainer;
