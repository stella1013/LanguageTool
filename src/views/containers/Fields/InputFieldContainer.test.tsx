import React from 'react';
import { shallow } from 'enzyme';
import InputFieldContainer from './InputFieldContainer';
/*
	* Factory function to create a Shallow Wrapper for the App Component
	* @function setup
	* @param {object} props - component props
	* @param {object} state - initial state
	* @returns {Shallow Wrapper}
	*/ 
	const setup = (props={}, state={})=>{
		const wrapper = shallow(<InputFieldContainer {...props}/>);
		if(state) {
			wrapper.setState(state);
		};
		return wrapper;
	}
describe('Input Field Container', () => {
	//const input = shallow(<InputFieldContainer />);
	const wrapper = setup();
	it('renders without crashing', () => {
		expect(wrapper).toMatchSnapshot();
	});
});
