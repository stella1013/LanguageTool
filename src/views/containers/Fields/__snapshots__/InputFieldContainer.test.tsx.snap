// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Field Container renders without crashing 1`] = `ShallowWrapper {}`;
